### PR TITLE
Feature/csv export button in charts

### PR DESCRIPTION
**Pull request by geclos**
- Adds csv export popover to all charts and tables
- Adds `download` to React's useQuery

### CODE REVIEW

File: packages/client/core/src/data/api.ts [hunk 1]
```diff
@@ -50,16 +54,28 @@ export class LatitudeApi {
...
+  }
+
+  async downloadQuery({
+    queryPath,
+    params = {},
+  }: {
+    queryPath: string
+    params?: QueryParams
+  }) {
+    return this.get<Blob>(`api/queries/${queryPath}`, {
```
Comment (andresgutgon) [inline packages/client/core/src/data/api.ts:69]: Dry use `getQuery<Blob>` just make `getQuery` generic and make `download` a flag

```javascript
return this.getQuery<Blob>({ queryPath, params, force, download: true })
```

```javascript
getQuery<T>: Promise<T>
```

Comment (andresgutgon) [reply packages/client/core/src/data/api.ts:69]: Because if not download can't be forced

File: packages/client/core/src/theme/ui/card/index.ts [hunk 2]
```diff
@@ -18,8 +17,8 @@ const cardHeaderVariants = tv({
   base: 'flex flex-col space-y-1.5',
   variants: {
     type: {
-      normal: NORMAL_PADDING,
-      invisible: 'pb-6',
+      normal: 'pb-6',
+      invisible: '',
```
Comment (andresgutgon) [inline packages/client/core/src/theme/ui/card/index.ts:21]: How looks this now?

File: packages/client/svelte/src/lib/internal/ui/chart/_Wrapper.svelte [hunk 3]
```diff
@@ -99,10 +103,45 @@
...
+                      <circle cx="8" cy="12" r="1" fill="currentColor" />
+                      <circle cx="12" cy="12" r="1" fill="currentColor" />
+                      <circle cx="16" cy="12" r="1" fill="currentColor" />
+                    </svg>
+                  </div>
+                </Popover.Trigger>
+                <Popover.Content class="flex px-0 py-2">
+                  <div class="w-full px-2 hover:bg-gray-100">
+                    <button on:click={download}>
+                      <Text size="h5">Download as CSV</Text>
```
Comment (andresgutgon) [inline packages/client/svelte/src/lib/internal/ui/chart/_Wrapper.svelte:136]: Yep i18n is a thing : /

File: packages/client/core/src/theme/ui/card/index.ts (hunk 2 above)
Comment (geclos) [reply packages/client/core/src/theme/ui/card/index.ts:21]: i've removed some extra padding we do not want

File: apps/server/src/lib/autoimports/chart/FunnelChart.svelte [hunk 4]
```diff
@@ -2,19 +2,32 @@
...
   import { DEFAULT_HEIGHT } from './constants'
+  import { api } from '@latitude-data/client'
+
+  type $$Props = Omit<
+    FunnelChartProps,
+    'download' | 'data' | 'isLoading' | 'error'
+  > &
+    QueryProps & {
+      download?: boolean
+    }
```
Comment (andresgutgon) [inline apps/server/src/lib/autoimports/chart/FunnelChart.svelte:14]: This is repeated everywhere. We could have a generic:

``` javascript
type VisualizationProps<T extends ChartProps> = Omit<T, 'download' | 'data' | 'isLoading' | 'error'> & {
  download?: boolean
}
```

**Review by andresgutgon — DISMISSED** [review]

**Pull request merged**